lib/rubocop/cop/style/lambda.rb

```diff
@@ -38,6 +38,19 @@ def lambda_length(block_node)
 
           end_line - start_line
         end
+
+        def autocorrect(node)
+          expr = node.ancestors.first.loc.expression
+
+          @corrections << lambda do |corrector|
+            src = if node.loc.expression.source == 'lambda'
+                    expr.source.sub(/^lambda\s+\{\s*\|([^\|]+)\|/, '->(\\1) {')
```

Comment:
Lambdas taking no arguments are not supported here, so they will be left unchanged. But the report will still say `[Corrected]`. I think the best solution would be to have a special message for the no arguments case: `Use the new lambda literal syntax -> {...}.`, and then to handle it in `autocorrect` as well.